example/router/router.hpp

```diff
@@ -0,0 +1,260 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_ROUTER_HPP
+#define BOOST_URL_ROUTER_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <detail/router.hpp>
+#include <memory>
+#include <vector>
+#include <stdexcept>
+
+namespace boost {
+namespace urls {
+
+/// Base class for all routers
+class router_base
+{
+    class impl;
+    std::unique_ptr<impl> impl_;
```

Comment:
You never used this... all of that node stuff and those private functions were supposed to be defined in the TU

Reply:
Do you mean "this" the `impl_` member? Everything related to `router_base::impl` is in the `.ipp` file. Where should it be?

Reply:
Yes `impl_` is unused. It would be better if all those protected and private declarations were in the .ipp file. We can discuss.